driver/hinge_embedding_loss_driver.hpp

```diff
@@ -0,0 +1,722 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef GUARD_MIOPEN_HINGE_EMBEDDING_LOSS_DRIVER_HPP
+#define GUARD_MIOPEN_HINGE_EMBEDDING_LOSS_DRIVER_HPP
+
+#include "InputFlags.hpp"
+#include "driver.hpp"
+#include "miopen/errors.hpp"
+#include "miopen/miopen.h"
+#include "tensor_driver.hpp"
+#include "tensor_view_5d.hpp"
+#include "timer.hpp"
+#include "random.hpp"
+#include <algorithm>
+#include <cfloat>
+#include <cstdlib>
+#include <memory>
+#include <numeric>
+#include <vector>
+#include <../test/tensor_holder.hpp>
+#include <../test/verify.hpp>
+
+#ifndef MLO_HINGE_EMBEDDING_LOSS_MHOST_H_
+#define MLO_HINGE_EMBEDDING_LOSS_MHOST_H_
+
+inline tensor_view_5d_t get_inner_expanded_tv(const miopen::TensorDescriptor Desc)
```

Comment:
This function is used repeatedly. I think it is a good idea to create src/inlude/loss/utils, define this function, and then use it by #include "../src/include/miopen/loss/utils.hpp".

Reply:
I have created utils.hpp file as you suggest